spec/git-repository-async-spec.coffee

```diff
@@ -0,0 +1,406 @@
+temp = require 'temp'
+GitRepositoryAsync = require '../src/git-repository-async'
+Git = require 'nodegit'
+fs = require 'fs-plus'
+os = require 'os'
+path = require 'path'
+Task = require '../src/task'
+Project = require '../src/project'
+
+# Clean up when the process exits
+temp.track()
+
+copyRepository = ->
+  workingDirPath = temp.mkdirSync('atom-working-dir')
+  fs.copySync(path.join(__dirname, 'fixtures', 'git', 'working-dir'), workingDirPath)
+  fs.renameSync(path.join(workingDirPath, 'git.git'), path.join(workingDirPath, '.git'))
+  fs.realpathSync(workingDirPath)
+
+openFixture = (fixture)->
+  GitRepositoryAsync.open(path.join(__dirname, 'fixtures', 'git', fixture))
+
+describe "GitRepositoryAsync", ->
+  repo = null
+
+  # beforeEach ->
+  #   gitPath = path.join(temp.dir, '.git')
+  #   fs.removeSync(gitPath) if fs.isDirectorySync(gitPath)
+  #
+  # afterEach ->
+  #   repo.destroy() if repo?.repo?
+
+  describe "@open(path)", ->
+
+    # This just exercises the framework, but I'm trying to match the sync specs to start
+    it "repo is null when no repository is found", ->
+      repo = GitRepositoryAsync.open(path.join(temp.dir, 'nogit.txt'))
+
+      waitsForPromise {shouldReject: true}, ->
+        repo.repoPromise
+
+      runs ->
+        expect(repo.repo).toBe null
+
+  describe ".getPath()", ->
+    # XXX HEAD isn't a git directory.. what's this spec supposed to be about?
+    xit "returns the repository path for a .git directory path", ->
+      # Rejects as malformed
+      repo = GitRepositoryAsync.open(path.join(__dirname, 'fixtures', 'git', 'master.git', 'HEAD'))
+
+      onSuccess = jasmine.createSpy('onSuccess')
+
+      waitsForPromise ->
```

Comment:
If you also write the specs in ES6, `async/await` syntax could really clean up these async tests...

```js
it('does something async', function () {
  waitsForPromise(async function() {
    expect(await repo.getPath()).toBe('/some/path')
  })
})
```